app/src/main/java/ch/epfl/skysync/database/tables/LocationTable.kt

```diff
@@ -0,0 +1,55 @@
+package ch.epfl.skysync.database.tables
+
+import ch.epfl.skysync.database.FirestoreDatabase
+import ch.epfl.skysync.database.Table
+import ch.epfl.skysync.database.schemas.LocationSchema
+import ch.epfl.skysync.models.location.Location
+import com.google.firebase.firestore.ListenerRegistration
+import kotlinx.coroutines.CoroutineScope
+import com.google.firebase.firestore.Filter
+import kotlinx.coroutines.launch
+
+
+class LocationTable(
+    db: FirestoreDatabase,
+) : Table<Location, LocationSchema>(db, LocationSchema::class, PATH) {
+
+    /**
+     * Update the location of a user in the database.
+     *
+     * @param location The location to update.
+     * @param onError Optional error handler for catching exceptions.
+     */
+    suspend fun updateLocation(location: Location, onError: ((Exception) -> Unit)? = null) {

```

Comment:
It's a good idea to always update the location instead of creating a new one each time, however we need to choose what ID to set for the location, a very straightforward solution would be to directly use the user ID (thus removing the `userId` field), as we expect to store only one location for a user at any given time.

Reply:
yep